ui/src/app/cohort-search/selection-list/selection-list.component.tsx

```diff
@@ -310,111 +329,143 @@ export class SelectionListModalVersion extends React.Component<Props> {
-  }
+    removeCriteria(criteriaToDel) {
+      const updateList = fp.remove((selection) => selection.parameterId === criteriaToDel.parameterId, this.props.criteria);
+      currentCohortCriteriaStore.next(updateList);
+    }
+
+    renderCriteriaGroup(criteriaGroup, header) {
+      return  <React.Fragment>
+        <h3> {header}</h3>
+        <hr style={{marginRight: '0.5rem'}}/>
+        {criteriaGroup && criteriaGroup.map((criteria, index) =>
+          <SelectionInfo key={index}
+                         index={index}
+                         selection={criteria}
+                         removeSelection={() => this.removeCriteria(criteria)}/>
 
-  applyModifier(modifiers) {
-    if (modifiers) {
-      const modifierButtonText = '(' + modifiers.length + ')  MODIFIERS APPLIED';
-      this.setState({showModifiersSlide: false, modifierButtonText: modifierButtonText});
-    } else {
-      this.setState({showModifiersSlide: false, modifierButtonText: 'APPLY MODIFIERS'});
+        )}
+      </React.Fragment> ;
     }
-  }
 
-  get showModifierButton() {
-    const {criteria} = this.props;
-    return criteria && criteria.length > 0 &&
-      criteria[0].domainId !== DomainType.PHYSICALMEASUREMENT.toString()
-      && criteria[0].domainId !== DomainType.PERSON.toString();
-  }
+    applyModifier(modifiers) {
+      if (modifiers) {
+        const modifierButtonText = '(' + modifiers.length + ')  MODIFIERS APPLIED';
+        this.setState({showModifiersSlide: false, modifierButtonText: modifierButtonText});
+      } else {
+        this.setState({showModifiersSlide: false, modifierButtonText: 'APPLY MODIFIERS'});
+      }
+    }
 
-  render() {
-    const {back, criteria} = this.props;
-    const {modifierButtonText, showModifiersSlide} = this.state;
-    return <div>
-        {!showModifiersSlide ?  <React.Fragment>
-          <h3 style={{...styles.sectionTitle, marginTop: 0}}>Add selected criteria to cohort</h3>
-          <div style={{paddingTop: '0.5rem', position: 'relative'}}>
-            <div style={styles.selectionContainer}>
-              {this.renderCriteria()}
-              {this.showModifierButton && <div style={{paddingLeft: '0.6rem'}}>
-                <Button type='secondaryOnDarkBackground' style={styles.modifierButton}
-                        onClick={() => this.setState({showModifiersSlide: true})}>
-                  {modifierButtonText}
-                </Button>
-              </div>}
+    get showModifierButton() {
+      const {criteria} = this.props;
+      return criteria && criteria.length > 0 &&
+        criteria[0].domainId !== DomainType.PHYSICALMEASUREMENT.toString()
+        && criteria[0].domainId !== DomainType.PERSON.toString();
+    }
+
+    render() {
+      const {back, criteria} = this.props;
+      const {attributesSelection, modifierButtonText, showModifiersSlide} = this.state;
+      return <div>
+        <FlexRow style={styles.navIcons}>
+          {(attributesSelection || showModifiersSlide) &&
```

Comment:
nit we can move this in a function as we are using this condition twice (at 442) attributesSelection || showModifiersSlide